### ISSUE TITLE
Running yes-or-no-p still hangs emacs after 5b9f4b0

### ISSUE THREAD

**Issue by setzer22** [ch11ng/exwm#663]
This was supposedly fixed in commit 5b9f4b0, which has been introduced in the current release of exwm (0.23). However, regular `yes-or-no-p` questions still hang emacs for me. 

To reproduce, I can use this snippet:
```
(exwm-input-set-key (kbd "s-a") (lambda () (interactive) (yes-or-no-p "TEST")))
```
If this runs during emcas startup, i can press Super+a to trigger a simple yer-or-no-p in the minibuffer. Doing this when an EXWM buffer in char mode is focused hangs emacs just like before commit 5b9f4b0. 

I have noticed an improvement after the commit, though: When exwm asks me to kill an unresponsive application while I'm in a focused char-mode window, I can do so without emacs freezing.

**Comment by ch11ng** [comment]
Could you prepare a debug log? I don't see any problem with `yes-or-no-p` in char-mode here.

**Comment by setzer22** [comment]
Sure, here's a debug log: https://pastebin.com/n8yQ4WT5

I tried to keep it as short as possible, but the nature of the issue makes it difficult for me to focus back the `*XELB-DEBUG*` buffer after I reproduce it. This is what I did:
 
- I was originally in the `*XELB-DEBUG*` buffer, I switched to my firefox window.
- I pressed `s-a`, which is bound to the function in my first post
- This requested input from the mini-buffer. However, pressing keys at that point sent them directly to firefox
- To recover from this, I used plasma's workspace switcher to go to a workspace where a non-exwm buffer was focused (this is the only way I can recover from this state)
- Finally, after I managed to focus a non-exwm buffer, the mini-buffer took my input and yes-or-no-p finished.

**Comment by ch11ng** [comment]
It seems you are on an old release as the debug log contains no timestamp.  (Or you disabled it?)

**Comment by setzer22** [comment]
I didn't (consciously) disable anything :S. Actually, I haven't added any new configuration for exwm since the update. 

If I run package-list-packages, exwm displays as 0.23. I am not very knowledgeable about how emacs handles packages, I just did a general package update through spacemacs, but I guess if I see 0.23 it means I have the correct version, right?

**Comment by ch11ng** [comment]
Please try <kbd>C-k exwm-init RET</kbd> and follow the link to see where the source is located.

**Comment by setzer22** [comment]
I'm currently on spacemacs (so no regular emacs bindings). Still, I think `C-k` is used to kill line in emacs, isn't it? What function is `C-k` supposed to invoke in your example?

**Comment by ch11ng** [comment]
Sorry I meant <kbd>C-h k</kbd>.

**Comment by setzer22** [comment]
It seems `C-h k` is `describe-key` for me. But anyway, I found out how to do it in my emacs. You were right, I checked the documentation for the exwm-init symbol and if was pointing to a folder called exwm-0.22 for some reason.

It seems this is an error in my configuration. Perhaps I have multiple exwm packages installed... I will try to fix this on my side. Once I verify the issue was that I'll close this.

Thanks!
